engine.py

```diff
@@ -72,6 +72,13 @@ def destroy_engine(self):
         if hasattr(self.__impl, "destroy_engine"):
             self.__impl.destroy_engine()
 
+    def _emit_log_message(self, handler, record):
```

Comment:
The desktop engine is broken in two halves. The part in engine.py, which is the base implementation common to the Shotgun Desktop and its background process and the `desktop_engine_{site,project}_implementation.py` file. The __impl is the implementation coming from that second file. Desktop engine's custom handlers are implemented in those files and when the core calls these methods they are forwarded to the appropriate location.